inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.cpp

```diff
@@ -745,18 +774,392 @@ class CompareNodesAttributes {
-    attr_comparison::ReadAndCompareAttributes m_compare_attr;
+    ReadAndStoreAttributes m_store_attr;
+    ReadAndCompareAttributes m_compare_attr;
+};
+
+}  // namespace detail
+
+Comparator::Result compare(
+    ngraph::Node* node1, ngraph::Node* node2, Comparator::CmpValues comparition_flags) {
+    detail::CompareNodesAttributes compare_nodes_attr(comparition_flags);
+    node1->visit_attributes(compare_nodes_attr.get_ref_reader());
+    node2->visit_attributes(compare_nodes_attr.get_cmp_reader());
+    if (!compare_nodes_attr.equal()) {
+        return Comparator::Result::error(
+            "Comparison of attributes failed for nodes " + name(node1) + ", " + name(node2) +
+            " [cmp status: " + to_str(compare_nodes_attr) + "]");
+    }
+    return Comparator::Result::ok(to_str(compare_nodes_attr));
+}
+
+}  // namespace attributes
+
+namespace subgraph {
+
+namespace detail {
+
+template <typename Ptr>
+Ptr not_null(Ptr&& p) {
+    if (!p) {
+        THROW_IE_EXCEPTION << "empty pointer";
+    }
+    return std::forward<Ptr>(p);
+}
+
+class NodeAndInputDescription {
+public:
+    using SubGraphOp = ngraph::op::util::SubGraphOp;
+    using InputDescripton = SubGraphOp::InputDescription;
+    using InputNode = ngraph::Input<ngraph::Node>;
+    using Parameter = ngraph::opset6::Parameter;
+
+    explicit NodeAndInputDescription(
+        const InputNode& input, const Parameter* parameter, const InputDescripton* description)
+        : input(input), parameter(not_null(parameter)), description(not_null(description)) {}
+
+    static bool equal_descriptions(const InputDescripton* lhs, const InputDescripton* rhs) {
+        if (!lhs || !rhs || lhs->get_type_info() != rhs->get_type_info()) {
+            return false;
+        }
+
+        if (lhs->get_type_info() == SubGraphOp::SliceInputDescription::type_info) {
+            using InDesc = SubGraphOp::SliceInputDescription;
+            const InDesc* l_input = static_cast<const InDesc*>(lhs);
+            const InDesc* r_input = static_cast<const InDesc*>(rhs);
+            return l_input->m_start == r_input->m_start && l_input->m_stride == r_input->m_stride &&
+                   l_input->m_part_size == r_input->m_part_size &&
+                   l_input->m_end == r_input->m_end && l_input->m_axis == r_input->m_axis;
+        } else if (lhs->get_type_info() == SubGraphOp::MergedInputDescription::type_info) {
+            return true;  // noting extra to check
+        } else if (lhs->get_type_info() == SubGraphOp::InvariantInputDescription::type_info) {
+            return true;  // noting extra to check
+        }
+
+        THROW_IE_EXCEPTION << "Type is not supported: [" << lhs->get_type_info().name << "]";
+
+        return true;
+    }
+
+    bool parameter_and_input_match() const {
+        if (description->get_type_info() == SubGraphOp::SliceInputDescription::type_info) {
+            ///
+            /// TODO shape calculation - required decent formula to check size from Slice
```

Comment:
Is any function to check this or should I copy formula from `CoordinateTransform`:
```
    for (size_t axis = 0; axis < m_n_axes; axis++)
    {
        m_target_shape.push_back(ceil_div(source_end_corner[source_axis_order[axis]] -
                                              source_start_corner[source_axis_order[axis]],
                                          source_strides[source_axis_order[axis]]));
    }
```